### PR TITLE
showreplies added on new user creation #140

### PR DESCRIPTION
**Pull request by coleGillespie**
when a new user gets created make show replies the default so that the check boxes are checked on both profile and posts page

### CODE REVIEW

**Comment by coleGillespie** [comment]
@isaacs something like this?

**Comment by coleGillespie** [comment]
@jasonrhodes i actually forgot about commit amend. great feature and would have come in handy earlier. will use it more going forward. thanks for reminding me.

**Comment by isaacs** [comment]
Looks good!  Needs a test, though.  Probably the best way to test this would be to *remove* the test that sets the user-setting in test.js.

**Comment by coleGillespie** [comment]
@isaacs maybe a better way would be to make sure that after a new user is created the default is checked?

we should keep the test that sets the user setting imo since that is still a valid test

**Comment by isaacs** [comment]
So, the bottom 4 test cases right now do the following:

1. Explicitly set the checkbox for the user.
2. Verify that the box is checked on the new post page.
3. Explicitly unset the checkbox for the user.
4. Verify that the box is unchecked on the new post page.

If you remove the first test, *without* your change, then it'll fail on step 2.  But *with* your change, it'll pass.

Adding that other test you suggest is also good, though :)

**Comment by isaacs** [comment]
That is, with your change, step 1 isn't doing anything.

File: test/test.js [hunk 1]
```diff
@@ -730,6 +730,24 @@ lab.test('post page defaults to showing replies', function (done) {
   });
 });
 
+lab.test('profile page defaults to showing replies', function (done) {
```
Comment (isaacs) [inline test/test.js]: In order to test that the option is set immediately on new user creation, it should move up a few steps, to above line 690.  It should replace the test on line 690, in fact.

**Comment by coleGillespie** [comment]
test seems to be failing to find the checked checkbox after a user is created. trying to determine why.

**Comment by coleGillespie** [comment]
@isaacs I needed to move the test above the first interaction with the profile. During the test on line 231 when add a name ( and other things ) but we do not set the showreplies flag to true. This is causing it to be set to false which is why my test was failing before it.

**Comment by isaacs** [comment]
@coleGillespie Awesome!

So, what we have now is, we verify that it's set, then we (inadvertently, whatever) unset it, then set it again later, etc.  Love it.

**Pull request merged**